### PR TITLE
Test command completion requires symfony/console 5.4+

### LINKED ISSUE #43594: [DX] Add completion to Symfony commands
Completion of command arguments and options is coming in Symfony 5.4 (#38275). Command and option names are already autocompleted, but argument and option values needs to be implemented on each command.

As an example, [#42251](https://github.com/symfony/symfony/pull/42251/files#diff-771e524eaa21e31d32047fda540ba0b53222c74e693bee5a45f6da7fa1644571R87-R96) added completion to  `secrets:remove`.

This feature needs to be implemented on Symfony commands that has predictable values for their arguments and options.

Some guidelines:

- Don't complete file/directory paths for now. See https://github.com/symfony/symfony/issues/43594#issuecomment-947385485
- You don't need to filter the suggestions in PHP, `compgen` (used by the shell script) does this for you based on the suggestions provided by PHP
- You can use the debugging utility while working on providing suggestions. See https://github.com/symfony/symfony/issues/43594#issuecomment-947535113

**PLEASE, comment here BEFORE starting to work on something and WAIT for the confirmation to avoid duplicate work.**

*All commands have been claimed*

 * [x] Command `debug:event-dispatcher` (`event`, `--dispatcher`, `--format`) @stephenkhoo https://github.com/symfony/symfony/pull/43859
 * [x] Command `debug:config` (`name`, `path`) @eclairia https://github.com/symfony/symfony/pull/43850
 * [x] Command `debug:autowiring` (`search`) @eclairia https://github.com/symfony/symfony/pull/43838
 * [x] Command `messenger:failed:remove` (`id`, `--transport`) @scyzoryck https://github.com/symfony/symfony/pull/43663
 * [x] Command `messenger:failed:retry` (`id`, `--transport`) @scyzoryck https://github.com/symfony/symfony/pull/43663
 * [x] Command `messenger:failed:show` (`id`, `--transport`) @scyzoryck https://github.com/symfony/symfony/pull/43663
 * [x] Command `debug:container` (`name`) @GromNaN https://github.com/symfony/symfony/pull/43857
 * [x] Command `messenger:consume` (`receivers`, `--bus`, `--queues`) @GromNaN https://github.com/symfony/symfony/pull/43891
 * [x] Command `debug:firewall` (`name`, `--format`) @IonBazan https://github.com/symfony/symfony/pull/43598
 * [x] Command `debug:form` (`name`, `--format`) @IonBazan https://github.com/symfony/symfony/pull/43598
 * [x] Command `debug:messenger` (`bus`) @IonBazan https://github.com/symfony/symfony/pull/43598
 * [x] Command `debug:router` (`name`) @IonBazan https://github.com/symfony/symfony/pull/43598
 * [x] Command `cache:pool:clear` (`pools`) @andyexeter https://github.com/symfony/symfony/pull/43621
 * [x] Command `cache:pool:delete` (`pool`, `key`) @andyexeter https://github.com/symfony/symfony/pull/43621
 * [x] Command `completion` (`shell`) @dkarlovi https://github.com/symfony/symfony/pull/43615
 * [x] Command `config:dump-reference` (`name`, `path`) @StaffNowa https://github.com/symfony/symfony/pull/43682
 * [x] Command `help` (`command_name`, `--format`) @GromNaN https://github.com/symfony/symfony/pull/43596
 * [x] Command `debug:translation` (`locale`, `bundle`, `--domain`) @alexandre-daubois https://github.com/symfony/symfony/pull/43644
 * [x] Command `debug:twig` (`name`, `--filter`, `--format`) @StaffNowa https://github.com/symfony/symfony/pull/43846
 * [x] Command `list` (`namespace`, `--format`) @GromNaN https://github.com/symfony/symfony/pull/43596
 * [x] Command `lint:twig` (`filename`, `--format`) @makraz https://github.com/symfony/symfony/pull/43680
 * [x] Command `lint:xliff` (`filename`, `--format`) @makraz https://github.com/symfony/symfony/pull/43680
 * [x] Command `lint:yaml` (`filename`, `--format`, `--exclude`) @makraz https://github.com/symfony/symfony/pull/43680
 * [x] Command `messenger:setup-transports` (`transport`) @Tayfun74 https://github.com/symfony/symfony/pull/43640
 * [x] Command `secrets:remove` (`name`) @GromNaN https://github.com/symfony/symfony/pull/43626
 * [x] Command `secrets:set` (`name`, `file`) @GromNaN https://github.com/symfony/symfony/pull/43626
 * [x] Command `security:encode-password` (`user-class`) @94noni https://github.com/symfony/symfony/pull/43653
 * [x] Command `security:hash-password` (`user-class`) @94noni https://github.com/symfony/symfony/pull/43653
 * [x] Command `server:dump` (`--format`) @alexandre-daubois https://github.com/symfony/symfony/pull/43683
 * [x] Command `translation:pull` (`provider`, `--domains`, `--locales`, `--format`) @welcoMattic https://github.com/symfony/symfony/pull/43672 
 * [x] Command `translation:push` (`provider`, `--domains`, `--locales`) @welcoMattic https://github.com/symfony/symfony/pull/43672 
 * [x] Command `translation:update` (`locale`, `bundle`, `--format`, `--domain`, `--sort`) @stephenkhoo https://github.com/symfony/symfony/pull/43676
 * [x] Command `ulid:generate` (`--format`) @StaffNowa https://github.com/symfony/symfony/pull/43639
 * [x] Command `uuid:generate` (`--format`) @StaffNowa https://github.com/symfony/symfony/pull/43639
 * [x] Command `workflow:dump` (`name`, `marking`, `--label`, `--format`) @StaffNowa StaffNowa https://github.com/symfony/symfony/pull/43848

Commands that does not require auto-completion:

* [x] Command `about`
* [x] Command `assets:install` (`target`) (directory completion)
* [x] Command `cache:clear`
* [x] Command `cache:pool:list`
* [x] Command `cache:pool:prune`
* [x] Command `cache:warmup`
* [x] Command `debug:dotenv`
* [x] Command `dotenv:dump`
 * [x] Command `debug:validator` (`class`) @Kocal *completion support not yet possible*
* [x] Command `lint:container`
* [x] Command `messenger:stop-workers`
* [x] Command `router:match` (`path_info`)
* [x] Command `secrets:decrypt-to-local`
* [x] Command `secrets:encrypt-from-local`
* [x] Command `secrets:generate-keys`
* [x] Command `secrets:list`
* [x] Command `server:log` (`--host`, `--format`) @StaffNowa https://github.com/symfony/symfony/pull/43892
* [x] Command `ulid:inspect`
* [x] Command `uuid:inspect`

### PR DESCRIPTION
**Pull request by StaffNowa**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? |no
| Tickets       | Fix #43594
| License       | MIT
| Doc PR        | -

Test command completion requires symfony/console 5.4+

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Tests/Command/CachePoolClearCommandTest.php [hunk 1]
```diff
@@ -35,6 +35,10 @@ protected function setUp(): void
      */
     public function testComplete(array $input, array $expectedSuggestions)
     {
+        if (!class_exists(CommandCompletionTester::class)) {
+            $this->markTestSkipped('Test command completion requires symfony/console 5.4+.');
+        }
```
Comment (derrabus) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Command/CachePoolClearCommandTest.php]: FrameworkBundle already defines `"symfony/console": "^5.4|^6.0"` in its composer.json. This check should not be necessary.

File: src/Symfony/Component/Console/Tests/Command/DumpCompletionCommandTest.php [hunk 2]
```diff
@@ -13,6 +13,10 @@ class DumpCompletionCommandTest extends TestCase
      */
     public function testComplete(array $input, array $expectedSuggestions)
     {
+        if (!class_exists(CommandCompletionTester::class)) {
+            $this->markTestSkipped('Test command completion requires symfony/console 5.4+.');
+        }
```
Comment (derrabus) [inline src/Symfony/Component/Console/Tests/Command/DumpCompletionCommandTest.php]: This test case is part of the `symfony/console` package. We should not need to check if a class from the same package is present.

**Comment by StaffNowa** [comment]
@derrabus thanks only two components left :)

**Comment by derrabus** [comment]
> @derrabus thanks only two components left :)

If you look at the composer.json files of those two components, you will find that they have the same `require-dev` constraint as FrameworkBundle. So basically, this PR does nothing. Can you elaborate what problem you wanted to solve?

**Comment by StaffNowa** [comment]
> > @derrabus thanks only two components left :)
> 
> If you look at the composer.json files of those two components, you will find that they have the same `require-dev` constraint as FrameworkBundle. So basically, this PR does nothing. Can you elaborate what problem you wanted to solve?

Just got message about validation making and checked all code :) Thanks, I will close this PR

**Pull request closed without merging**